rfcs/0041-selinux-support.md

```diff
@@ -0,0 +1,159 @@
+---
+feature: selinux-support
+start-date: 2019-03-10
+author: Alexander Kahl (@e-user)
+co-authors: (none yet)
+related-issues: NixOS/nix#2374, NixOS/nix#2670, NixOS/nixpkgs#56965
+---
+
+[NixOS/nix#2374]: https://github.com/NixOS/nix/issues/2374
+[NixOS/nix#2670]: https://github.com/NixOS/nix/pull/2670
+[NixOS/nixpkgs#56965]: https://github.com/NixOS/nixpkgs/pull/56965
+
+# Summary
+[summary]: #summary
+
+Add support for [SELinux] to Nix and NixOS. This means that Nix will be
+able to write SELinux file contexts on enabled systems and NixOS will be
+provided with a base policy that supports as many use cases as possible. New
+options in NixOS make it possible to build Linux with SELinux support enabled.
+
+[SELinux]: https://selinuxproject.org/page/Main_Page
+
+# Motivation
+[motivation]: #motivation
+
+Security Enhanced Linux is a Linux kernel security feature providing mandatory
+access controls. It confines processes to access files with corresponding labels
+for specific purposes, such as reading and writing.
+
+There is currently one family of Linux distributions with SELinux enforcing
+active by default: [Fedora] and derivatives, which includes [Red Hat Enterprise
+Linux], [CentOS], Scientific Linux and others. Several other Linux distributions
+have optional support for SELinux, including [Arch], [Debian], [Ubuntu], CoreOS,
+and [Gentoo].
+
+Without explicit file contexts written to files in the Nix store and Nix
+profiles, Nix-provided daemons currently cannot integrate with SELinux-enforcing
+systems, as trying to run them from systemd or interacting with them with
+systemctl results in kernel-level denials. This includes related issue
+[NixOS/nix#2374].
```

Comment:
As I understand it, the real motivation of this RFC is to allow to run Nix on SELinux-enabled distributions. This is what I am inferring here but it might be worth writing it down explicitly.

Reply:
That was also my original motivation, indeed - it's however just another step from there to full SELinux support in NixOS, which is why the RFC proposes to be split up for the second step, as an alternative.

Reply:
@zimbatm does the new addition help to clarify the motivation?

Reply:
thanks it does